### PR TITLE
raft: bcast when has pending conf change

### PR DESCRIPTION
**Pull request by BusyJay**
In some cases, leader can be removed from cluster. If commit is not broadcast, follower will still consider leader is included in the cluster which may affect the quorum calculation. If another node is isolated, election may not succeed any more.

This pr forces broadcasting commit when conf change is detected.

### CODE REVIEW

File: src/raft/raft.rs [hunk 1]
```diff
@@ -1412,7 +1412,7 @@ impl<T: Storage> Raft<T> {
         );
         if maybe_commit {
             if self.maybe_commit() {
-                if !self.skip_bcast_commit {
+                if self.should_bcast_commit() {
```
Comment (siddontang) [inline src/raft/raft.rs]: bcast in the remove node is enough?

**Comment by javaforfun** [comment]
maybe leader can reject the remove request.  use leader_transfer + remove_node instead?

**Comment by BusyJay** [comment]
This is not reliable. A peer can be follower when proposing the conf change but be leader when applying. And actually @disksing has done the check in best effort already.

**Comment by siddontang** [comment]
LGTM

PTAL @javaforfun

**Pull request merged**